scripts/perf2perfetto/README.md

```diff
@@ -0,0 +1,201 @@
+## Introduction
+
+Recent Intel processors feature the "Intel Processor Trace" feature, which can be used to capture the full execution trace of a program.
+It is an amazing tool for debugging, optimization and learning how (natively compiled) programs work.
+
+Linux supports Intel PT in `perf`.
+This repository contains an exporter of Intel PT traces from `perf` to [Fuchsia trace format](https://fuchsia.dev/fuchsia-src/reference/tracing/trace-format) for convenient viewing in [Perfetto](https://perfetto.dev/).
+
+## Example output (viewed in Perfetto)
+
+![Screenshot of an example output viewed](images/screenshot.png)
+
+## How to use
+
+### Basic usage
+
+First collect the trace with `perf record`. For example, to trace a running process `my_prog`, you can do:
+
+```bash
+TRACE_DURATION_SECS=1
+PID=$(pgrep --newest --exact my_prog)
+sudo perf record -o perf.data -p $PID -e intel_pt/cyc=1/ -- sleep $TRACE_DURATION_SECS
```

Comment:
Q: when you are saving hundreds of MiB/sec to your disk, isn't this workload competing with the workload you are trying to profile?

Reply:
Are you asking about how PT could be used in production?

Yes, the amount of data generated by PT is quite big, so it could be competing with the profiled workload, both for memory and for the disk. But the memory overhead is usually acceptable enough, and disk interference can be minimized by writing to a ramdisk. And I guess in most use cases the traces you need are short enough not to take that much disk.

Reply:
And, if it wasn't clear, this script/PR isn't about profiling performance in production, but about breaking down some small things (such as a single query) in a local test.

Reply:
Yes, I was thinking that the example should use -o /dev/shm/perf.data (with a warning attached)

Reply:
I don't actually expect *us* to use PT in production for anything, mainly because AFAIK most (almost all?) of our use cases run on VMs, and PT usually isn't available on anything that isn't bare metal.